src/main/java/org/apache/maven/shared/dependency/analyzer/asm/DependencyClassFileVisitor.java

```diff
@@ -71,7 +71,7 @@ public void visitClass(String className, InputStream in) {
             reader.accept(classVisitor, 0);
         } catch (IOException exception) {
             exception.printStackTrace();
-        } catch (IndexOutOfBoundsException e) {
+        } catch (IndexOutOfBoundsException | IllegalArgumentException e) {
```

Comment:
The comment below doesn't apply to the new exception. It's probably better to make this a spearate catch block with a more descriptive error message.

Reply:
Hi @elharo 
I rewrote the catch clause per your suggestion.